### PR TITLE
App content digest is changed after successful upload

### PR DESCRIPTION
**Pull request by ikasarov**
see LMCROSSITXSADEPLOY-2266

### CODE REVIEW

**Review by radoslav-d — APPROVED** [review]
Seems alright

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java [hunk 1]
```diff
@@ -75,7 +75,10 @@ public StepPhase executeAsyncStep(ProcessContext context) throws FileStorageExce
 
         boolean contentChanged = detectApplicationFileDigestChanges(context, cloudApp, newApplicationDigest);
         if (contentChanged) {
-            return proceedWithUpload(context, applicationToProcess, moduleFileName);
+            proceedWithUpload(context, applicationToProcess, moduleFileName);
+            attemptToUpdateApplicationDigest(context.getControllerClient(), cloudApp, newApplicationDigest);
+            setAppContentChanged(context, true);
```
Comment (boyan-velinov) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java:80]: Why not including setAppContentChanged in proceedWithUpload()?

**Review by boyan-velinov — COMMENTED** [review]
Can we add some unit tests here?

**Review by nictas — APPROVED** [review]

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java (hunk 1 above)
Comment (ikasarov) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/UploadAppStep.java]: done

**Pull request merged**